File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> numberVsFrequencyMap = new HashMap<>();
        for (int num: nums) {
            numberVsFrequencyMap.put(num, numberVsFrequencyMap.getOrDefault(num, 0) + 1);
        }
        List<Map.Entry<Integer, Integer>> keyValueEntryMapList = new ArrayList<>(numberVsFrequencyMap.entrySet());
        keyValueEntryMapList.sort((a, b) -> {
            return Integer.compare(b.getValue(), a.getValue());
        });
        List<Integer> resultList = new ArrayList<>();
        for (int i = 0; i < k; i++) {
            resultList.add(keyValueEntryMapList.get(i).getKey());
        }
        int[] result = new int[resultList.size()];
        for (int i = 0; i < resultList.size(); i++) {
            result[i] = resultList.get(i);
        }
        return result;
    }
}

File: Data Structures & Algorithms/majority-element-ii/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        dict = {}
        for num in nums:
            if num in dict:
                dict[num] += 1
            else:
                dict[num] = 1
        
        result = []
        for key, value in dict.items():
            if value > (len(nums) // 3):
                result.append(key)
        
        return result
        